source/server/listener_manager_impl.cc

```diff
@@ -0,0 +1,126 @@
+#include "server/listener_manager_impl.h"
+
+#include "envoy/registry/registry.h"
+
+#include "common/common/assert.h"
+#include "common/json/config_schemas.h"
+#include "common/network/listen_socket_impl.h"
+#include "common/network/utility.h"
+#include "common/ssl/context_config_impl.h"
+
+#include "server/configuration_impl.h"
+
+namespace Envoy {
+namespace Server {
+
+Network::ListenSocketPtr
+ProdListenSocketFactory::create(Network::Address::InstanceConstSharedPtr address,
```

Comment:
Basically just the addition of the `ProdListenSocketFactory` implementation and associated interface, as well as making some things const.